### PR TITLE
fix: trino sql runner tables

### PR DESCRIPTION
**Pull request by rephus**
Closes:  https://github.com/lightdash/lightdash/issues/11865

### Description:

Screenshot from 2024-10-15 11-30-20

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts [hunk 1]
```diff
@@ -331,15 +331,41 @@ export class TrinoWarehouseClient extends WarehouseBaseClient<CreateTrinoCredent
...
+                schema
+                    ? `AND table_schema = '${this.sanitizeInput(schema)}'`
+                    : ''
+            }
             ${
                 database
-                    ? `AND table_catalog = ${this.sanitizeInput(database)}`
+                    ? `AND table_catalog = '${this.sanitizeInput(database)}'`
                     : ''
-            };
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts:339]: surprisingly, this ; was breaking the SQL 

> 2024-10-15 11:25:11 [Lightdash] error: Handled error of type WarehouseQueryError on [GET] /api/v1/projects/5d073877-5af7-49b1-bdcf-ea7b249138e9/sqlRunner/fields line 9:57: mismatched input ';'. Expecting: '%', '*', '+', '-', '.', '/', 'AND', 'AT', 'EXCEPT', 'FETCH', 'GROUP', 'HAVING', 'INTERSECT', 'LIMIT', 'OFFSET', 'OR', 'ORDER', 'UNION', 'WINDOW', '[', '||', <EOF>

File: packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts [hunk 2]
```diff
@@ -331,15 +331,41 @@ export class TrinoWarehouseClient extends WarehouseBaseClient<CreateTrinoCredent
                    data_type
             FROM information_schema.columns
             WHERE table_name = '${this.sanitizeInput(tableName)}'
-            ${schema ? `AND table_schema = ${this.sanitizeInput(schema)}` : ''}
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts:334]: missing ' quotes around filters

File: packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts [hunk 3]
```diff
@@ -331,15 +331,41 @@ export class TrinoWarehouseClient extends WarehouseBaseClient<CreateTrinoCredent
...
         `;
         const { rows } = await this.runQuery(query, tags);
 
         return this.parseWarehouseCatalog(rows, convertDataTypeToDimensionType);
     }
+
+    async getAllTables() {
+        const databaseName = this.connectionOptions.catalog;
+        const whereSql = databaseName
+            ? `AND table_catalog = '${this.sanitizeInput(databaseName)}'`
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts:353]: trino client doesn't take arguments on the query (we can't use ?) so afaik the only way is to sanitize the input on the sql

**Review by joaoviana — APPROVED** [review]
image
 
 Works great!

**Pull request merged**